src/adapter/src/catalog/open.rs

```diff
@@ -1067,18 +1029,16 @@ impl Catalog {
     /// # Errors
     #[tracing::instrument(level = "info", skip_all)]
     async fn load_system_configuration(
-        &mut self,
+        state: &mut CatalogState,
+        txn: &mut Transaction<'_>,
```

Comment:
not for this PR, but what do you think about bundling `CatalogState` and `Transaction` together?

Reply:
I think that's a good idea.